Plugin/Sales/Order/View/Details.php

```diff
@@ -40,14 +40,17 @@ public function __construct(
     public function beforeSetLayout(View $subject)
     {
         $amwalOrderId = $subject->getOrder()->getAmwalOrderId();
-        $amwalClient = $this->amwalClientFactory->create();
 
         try {
             // check if the $amwalOrderId have -canceled suffix
+            if (!isset($amwalOrderId) {
```

Comment:
There is a missing `)`, but I would opt for the more generic check in this case:
```suggestion
            if (!$amwalOrderId) {
```